File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    let answer = [];
    const map = {}
    for (let key of keymap) {
        key.split("").map((v, i) => map[v] = (map[v] < i+1 ? map[v] : i+1))
    }
    
    for (let target of targets) {
        answer.push(target.split("").reduce((a, c) => a += map[c], 0) || -1)
    }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):  
    answer = []
    keys = {}
    for key in keymap:
        for i,v in enumerate(key):
            keys[v] = min(keys[v], i + 1) if v in keys else i + 1
            
    for i,target in enumerate(targets):
        total = 0
        for w in target:
            if w not in keys: 
                total = -1
                break
            total += keys[w]
        answer.append(total)
        
    return answer